shellscript/bash-linux/s3/bucket-lifecycle-operations/bucket-operations-test.sh

```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env bash
+
+<<METADATA-DO-NOT-REMOVE
+# * Copyright 2010-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# *
+# * This file is licensed under the Apache License, Version 2.0 (the "License").
+# * You may not use this file except in compliance with the License. A copy of
+# * the License is located at
+# *
+# * http://aws.amazon.com/apache2.0/
+# *
+# * This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+# * CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# * specific language governing permissions and limitations under the License.
+METADATA-DO-NOT-REMOVE
+
+function pause {
+    if [[ $interactive = true ]]; then
+        read -p "Press ENTER to continue..."
+    fi
+}
+
+function usage {
+    echo "This script tests Amazon S3 bucket operations in the AWS CLI."
+    echo "It creates a randomly named bucket, copies files to it, then"
+    echo "deletes the files and the bucket."
+    echo ""
+    echo "To pause the script between steps so you can see the results in the"
+    echo "AWS Management Console, include the parameter -i."
+}
+
+# Parse parameters
+    if [ $# -gt 1 ]
+    then
+        echo "Too many parameters."
+        usage
+        exit 1
+    fi
+
+    if [[ $# == 0 ]]; then
+        interactive=false
+    elif [[ "$1" == "-i" ]]; then
+        interactive=true
+    else
+        echo "Invalid parameter."
+        usage
+        exit 1
+   fi
+
+# Initialize the shell scripts RANDOM variable
```

Comment:
We don't need all of these comments. Just an initial one:

# Create three random values for the bucket name
# which will look like testbucket-1432-9475-4983

Reply:
I prefer being a bit more explicit when it might be confusing what i'm trying to do. Especially when it is not DIRECTLY related to the API I'm trying to demonstrate.